airflow/www/static/js/dags.js

```diff
@@ -126,16 +126,75 @@ $(".typeahead").typeahead({
       success: callback,
     });
   },
+  matcher(item) {
+    const it = this.displayText(item);
+    const query = this.query.toLowerCase();
+    const queryValue = query.replace(/^([a-z]+:\s*)/i, "");
+
+    return it.toLowerCase().indexOf(queryValue) !== -1;
+  },
+  displayText(item) {
+    if (typeof item !== "undefined" && typeof item.name !== "undefined") {
```

Comment:
Do we want to check against the string `"undefined"` or the javascript value `undefined`?